pkg/controller/postgressnapshot/postgressnapshot_controller.go

```diff
@@ -118,40 +118,40 @@ func (r *ReconcilePostgresSnapshot) Reconcile(request reconcile.Request) (reconc
 	if err != nil {
 		errMsg := fmt.Sprintf("failed to get postgres resource: %s", err.Error())
 		if updateErr := resources.UpdateSnapshotPhase(ctx, r.client, instance, croType.PhaseFailed, croType.StatusMessage(errMsg)); updateErr != nil {
-			return reconcile.Result{Requeue: true, RequeueAfter: resources.ErrorReconcileTime}, updateErr
+			return reconcile.Result{Requeue: true, RequeueAfter: r.provider.GetReconcileTime(instance)}, updateErr
 		}
-		return reconcile.Result{Requeue: true, RequeueAfter: resources.ErrorReconcileTime}, errorUtil.New(errMsg)
+		return reconcile.Result{Requeue: true, RequeueAfter: r.provider.GetReconcileTime(instance)}, errorUtil.New(errMsg)
 	}
 
 	// check postgres deployment strategy is aws
 	if !r.provider.SupportsStrategy(postgresCr.Status.Strategy) {
 		errMsg := fmt.Sprintf("the resource %s uses an unsupported provider strategy %s, only resources using the aws provider are valid", instance.Spec.ResourceName, postgresCr.Status.Strategy)
 		if updateErr := resources.UpdateSnapshotPhase(ctx, r.client, instance, croType.PhaseFailed, croType.StatusMessage(errMsg)); updateErr != nil {
-			return reconcile.Result{Requeue: true, RequeueAfter: resources.ErrorReconcileTime}, updateErr
+			return reconcile.Result{Requeue: true, RequeueAfter: r.provider.GetReconcileTime(instance)}, updateErr
 		}
-		return reconcile.Result{Requeue: true, RequeueAfter: resources.ErrorReconcileTime}, errorUtil.New(errMsg)
+		return reconcile.Result{Requeue: true, RequeueAfter: r.provider.GetReconcileTime(instance)}, errorUtil.New(errMsg)
 	}
 
 	if instance.DeletionTimestamp != nil {
 		msg, err := r.provider.DeletePostgresSnapshot(ctx, instance, postgresCr)
 		if err != nil {
 			if updateErr := resources.UpdateSnapshotPhase(ctx, r.client, instance, croType.PhaseFailed, msg.WrapError(err)); updateErr != nil {
-				return reconcile.Result{Requeue: true, RequeueAfter: resources.ErrorReconcileTime}, updateErr
+				return reconcile.Result{Requeue: true, RequeueAfter: r.provider.GetReconcileTime(instance)}, updateErr
```

Comment:
I think on error we do not want to specify a requeue time, as we want to keep reconciling on an error till we get passed it. Looking at the other controllers, we only slow down the reconcile after it has returned with no error, so that we can limit api calls to aws.

Maybe @aidenkeating can confirm this, if I am wrong we will need to stick a ticket in the backlog to update all the controllers requeue times
```suggestion
				return reconcile.Result{}, updateErr
```

Reply:
`RequeueAfter` will be ignored here

Reply:
I think we can just remove the `Result` body anywhere we return an non-nil `error`, I don't see this as anything major

Reply:
Thanks for the comments. I'll return an empty result body instead